### PR TITLE
Request For Review: Friendship Module 1 [Social Media Website Milestone 5]

### PR DESCRIPTION
**Pull request by rloterh**
- Created Friendship model and implemented the appropriate associations between  User, Friend, Friendship
- Authenticated users can send requests to other signed-up users, and see if their request is pending or accepted
- Authenticated users can receive requests from other signed-p users and choose to accept or decline
- Accepted Friends shows up in Friends list
- Test cases for this module are running correctly
- Other features for this milestone are implemented effectively

### CODE REVIEW

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @rloterh :clap: :clap:
#### You have done a good job :+1: The project is well implemented. However, there are a few requirements you need to implement before this project can be approved.

#### Enable linter:
- _Linter is not working in this Pull Request. Fix this by enabling the linter._

#### Application Behaviours:
- _Posts in the timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Currently the app display all the post to the logged-in user regardless the post's author is a friend of the logged-in user or not_

#### Tests:
- _Add more tests for the user model, controllers, and possibly integration test for the posts_

#### _Reviewed by Temitayo Adeleke_

**Comment by rloterh** [comment]
- Changes implemented effectively as per first reviewer.
- Kindly NOTE:
I've configured the the linters over and over but to no avail. Both rubocop and stylel
screen_1
screen_2
screen_3


int are working effectively on my local setup. But based on the issue found on this link: https://github.community/t/actions-not-working-correctly-for-forks/16649
I've trying all day with no success. Action enabled on my github, and my checked my configuration over and over and it's perfect. Please see the attached screenshots

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,41 +1,44 @@
...
+    <% if notice.present? %>
+      <div class="notice">
+        <p> <%= notice %> </p>
+      </div>
+    <% end %>
+    <% if alert.present? %>
+      <div class="alert">
+        <p> <%= alert %> </p>
+      </div>
+    <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:39]: - [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  

**The Resources below may come in handy**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
* [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hi @rloterh, You did wonderfully on your project, lovely 👏 

Could schedule a zoom meeting with any of the TSE to see if we can resolve the  GitHub actions thing? Also posting this issue on the Full-time student section of slack.

But before that, there are few things the last TSE may have missed out. Kindly endeavor to fix them even though they are [OPTIONAL] now.
- [ ] There should b a button to invite a user to friendship on the users list page 👍    and on single-user page  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- [ ] Your readme file can use some updates to reflect the job done on current milestone

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Comment by rloterh** [comment]
- solved main issues
- Invitation Button is already implemented on the users page (button is labelled Send Request)
- Other codes directly original. Will implement that during the last milestone

**Review by ebeagusamuel — APPROVED** [review]
# Project Approved :100: 
#### Hey there :wave: 
You have done well in this project, especially in getting the linter issue fixed.
You can go ahead and merge this branch and proceed with the next milestone.
Good luck

slack(@ebeagusamuel)

**Pull request merged**